src/main/webapp/scripts/itinerary-overview.js

```diff
@@ -1,54 +1,82 @@
-let map;
-let latitude;
-let longitude;
-
 function initMap() {
-    getUserLocation()
+    var directionsRenderer = new google.maps.DirectionsRenderer();
+    var directionsService = new google.maps.DirectionsService();
+
+    const heath = new google.maps.LatLng(32.8368128, -96.4820992);
 
-    map = new google.maps.Map(document.getElementById("map"), {
-        center: { lat: 32, lng: -96.4820992},
-        zoom: 10
+    var map = new google.maps.Map(document.getElementById("map"), {
+        zoom: 10,
+        center: heath
     });
+    directionsRenderer.setMap(map);
+
+    calculateAndDisplayRoute(directionsService, directionsRenderer);
+    document.getElementById("mode").addEventListener("change", function() {
+        calculateAndDisplayRoute(directionsService, directionsRenderer);
+        });
+    }
 
-    marker = new google.maps.Marker({
-        map:map,
-        position: { lat: 32, lng: -96.4820992},
+function dropMarker(lat,lng) {
+    const location = {lat: lat, lng: lng};
+    const marker = new google.maps.Marker({
+        map: map,
+        position: location, //Default coords if geolocation fails
         animation: google.maps.Animation.DROP,
 
-    });
-}
+        });
+    }
 
-function getUserLocation() {
-    const options = {timeout: 60000};
-    navigator.geolocation.getCurrentPosition(retrieveUserLatLng, errorHandler, options);
-}
+function calculateAndDisplayRoute(directionsService, directionsRenderer) {
+    const heath = new google.maps.LatLng(32.8368128, -96.4820992);
+    const summit = new google.maps.LatLng(33.0083762, -96.7793862);
 
-function retrieveUserLatLng(position) {
+    var selectedMode = document.getElementById("mode").value;
```

Comment:
Try to use `const`/`let` instead of var, since `var` isn't block-scoped and every time you declare one, it becomes shared with all JS files we have instead of being local to function scopes.